contrib/platform/src/com/sun/jna/platform/win32/Kernel32.java

```diff
@@ -4387,4 +4386,76 @@ Pointer VirtualAllocEx(HANDLE hProcess, Pointer lpAddress, SIZE_T dwSize,
+     *
+     * @return Returns the input language identifier if successful.
+     *         If the input identifier is nonzero, the function returns that value.
+     *         If the language identifier is 0, the function always succeeds and returns
+     *         the identifier of the language that best supports the Windows console.
+     *
+     * <p>
+     * When a thread is created, the thread user interface language setting is empty and the user interface for
+     * the thread is displayed in the user-selected language. This function enables the application to change
+     * the user interface language for the current running thread.
+     * </p>
+     * <p>
+     * Windows Vista and later: Calling this function and specifying 0 for the language identifier is identical
+     * to calling SetThreadPreferredUILanguages with the MUI_CONSOLE_FILTER flag set. If the application specifies
+     * a valid nonzero language identifier, the function sets a particular user interface language for the thread.
+     * </p>
+     *
+     * @see <a href="https://learn.microsoft.com/en-us/windows/win32/api/winnls/nf-winnls-setthreaduilanguage">SetThreadUILanguage</a>
+     */
+    int SetThreadUILanguage(int LangId);
+
+    /**
+     * Sets the thread preferred UI languages for the current thread.
+     * @see <a href="https://learn.microsoft.com/en-us/windows/win32/intl/user-interface-language-management">User Interface Language Management</a>
+     *
+     * @param dwFlags            Flags identifying format and filtering for the languages to set.
+     *
+     * <p>
+     * The following format flags specify the language format to use for the thread preferred UI languages.
+     * The flags are mutually exclusive, and the default is MUI_LANGUAGE_NAME.
+     * </p>
+     * <p>
+     * We recommend that you use MUI_LANGUAGE_NAME instead of MUI_LANGUAGE_ID.
+     * </p>
+     *
+     * @param pwszLanguagesBuffer Pointer to a double null-terminated multi-string buffer that contains an ordered,
+     *                            null-delimited list, in the format specified by dwFlags.
+     * <p>
+     * To clear the thread preferred UI languages list, an application sets this parameter to a null string or an empty
+     * double null-terminated string. If an application clears a language list, it should specify either a format flag
+     * or 0 for the dwFlags parameter.
+     * </p>
+     *
+     * @param pulNumLanguages     Pointer to the number of languages that the function has set in the thread preferred UI languages list.
+     *                            When the application specifies one of the filtering flags, the function must set this parameter to NULL.
+     *
+     * @return Returns TRUE if the function succeeds or FALSE otherwise.
+     */
+    boolean SetThreadPreferredUILanguages(DWORD dwFlags, StringArray pwszLanguagesBuffer, ULONGByReference pulNumLanguages);
```

Comment:
A `DWORD` is a 32bit integer. If there is no reason against it, please use a plain java `int`. For the `pwszLanguagesBuffer` please test if using a `String[]` works. The TypeMapper should handle the wrapping. For the final parameter `pulNumLanguages` I suggest to use a `IntByReference`.